src/leo-server-app/src/messageHandler.ts

```diff
@@ -0,0 +1,48 @@
+import { WebSocket } from "ws";
+
+let clientSocket: any = null; // This will hold the active WebSocket connection
+
+// Function to set the client socket
+function setClientSocket(socket: any) {
+  clientSocket = socket;
+}
+
+// Send data to the client and wait for the response
+function sendDataToClientAndAwaitResponse(dataToSend: any, timeout: number) {
+  return new Promise((resolve, reject) => {
+    let responseData = "";
+
+    // Timeout period when reading response data line by line
+    let fragmentTimeout: any = null;
+    const fragmentTimeoutDuration = 1000;
+
+    if (!clientSocket || clientSocket.readyState !== WebSocket.OPEN) {
+      resolve("WebSocket connection is not open");
+      return;
+    }
+
+    // Create a timeout to reject the promise if no response is received within the specified period
+    const timeoutId = setTimeout(() => {
+      resolve("Response timeout");
+    }, timeout);
```

Comment:
does this not need fragmentTimeOutduration as a parameter?

Reply:
no

Reply:
since this is timeout after sending data, fragmentTimeOutduration is used when receiving the response from the client